host/sysfs/spi_test.go

```diff
@@ -269,12 +269,68 @@ func TestSPI_Connect_Half(t *testing.T) {
 	}
 }
 
-func TestSPIIOCTX(t *testing.T) {
-	if v := spiIOCTx(1); v != 0x40206B00 {
-		t.Fatalf("Expected 0x40206B00, got 0x%08X", v)
+func TestSPIIOC(t *testing.T) {
+	testCases := []struct {
+		name          string
+		got, expected uint
+	}{
+		{"spiIOCMode", spiIOCMode, 0x40016B01},
+		{"spiIOLSBFirst", spiIOLSBFirst, 0x40016B02},
+		{"spiIOCBitsPerWord", spiIOCBitsPerWord, 0x40016B03},
+		{"spiIOCMaxSpeedHz", spiIOCMaxSpeedHz, 0x40046B04},
+		{"spiIOCMode32", spiIOCMode32, 0x40046B05},
 	}
-	if v := spiIOCTx(9); v != 0x41206B00 {
-		t.Fatalf("Expected 0x41206B00, got 0x%08X", v)
+
+	for _, tc := range testCases {
+		if tc.got != tc.expected {
+			t.Errorf("%s = 0x%x, expected 0x%x", tc.name, tc.got, tc.expected)
+		}
+	}
+}
+
+func TestSPIIOCTx(t *testing.T) {
```

Comment:
That test should fail on mips, right? We should delete this then, because anyway we have gohci workers that ensure that SPI works for real.

Reply:
You're right, they do fail. I verified this by cross-compiling the test and then running it on the target. I wasn't sure how to test this. Should we include arch-specific unit tests or just rely on the gohci tests?

Reply:
No need to test for the constants, integration testing will suffice here. Maybe the only thing that would be cool is to add a test to [spismoketest](https://github.com/google/periph/blob/master/cmd/periph-smoketest/spismoketest/spismoketest.go): read the values back via an ioctl on the file handle to confirm that the values were setup properly; that's the reason why the "read value back to confirm" code was written, as a long time ago the code was trying to write value and was silently failing (oops).

Reply:
FTR, I don't ask you to do the spismoketest check here, what you did is already great.

I still think the dead code and const checks should be removed.

Reply:
I added a check to read back the mode flag since that's the only flag we set during initialization. `SPI_IOC_MESSAGE` doesn't have a corresponding _IOR ioctl, so we can't verify the message ioctl.
I'm unsure if the `GetFlag` approach is the right one here, what do you think?